src/command_modules/azure-cli-resource/azure/cli/command_modules/resource/custom.py

```diff
@@ -169,20 +168,20 @@ def validate_arm_template(resource_group_name, template_file_path,
         :param str template_file_path:path to deployment template JSON file
         :param str parameters_file_path:path to deployment parameters JSON file
     '''
-    return _deploy_arm_template_core(resource_group_name, 'deployment_dry_run', template_file_path,
+    return _deploy_arm_template_core(resource_group_name, template_file_path, 'deployment_dry_run',
                                      parameters_file_path, mode, validate_only=True)
 
-def _deploy_arm_template_core(resource_group_name, deployment_name, template_file_path,
+def _deploy_arm_template_core(resource_group_name, template_file_path, deployment_name=None,
                               parameters_file_path=None, mode='incremental', validate_only=False):
     from azure.mgmt.resource.resources.models import DeploymentProperties
 
     parameters = None
     if parameters_file_path:
-        parameters = get_file_json(parameters_file_path)
+        parameters = _load_json(parameters_file_path)
```

Comment:
This is the logic the @ syntax would render obsolete. However, I don't think you'd be able to use the template file name in the deployment name if you do this because the parser will have already consumed it.